server/src/main/java/org/elasticsearch/common/lucene/uid/PerThreadIDVersionAndSeqNoLookup.java

```diff
@@ -108,7 +108,27 @@ public DocIdAndVersion lookupVersion(BytesRef id, LeafReaderContext context)
             if (versions.advanceExact(docID) == false) {
                 throw new IllegalArgumentException("Document [" + docID + "] misses the [" + VersionFieldMapper.NAME + "] field");
             }
-            return new DocIdAndVersion(docID, versions.longValue(), context.reader(), context.docBase);
+            final long seqNo;
+            final long term;
+            if (loadSeqNo) {
+                NumericDocValues seqNos = context.reader().getNumericDocValues(SeqNoFieldMapper.NAME);
```

Comment:
we should clean this up in master once backported. can you put a comment in there?